Nnlops hj

lenzip opened the pull request:
Fix for running JHUGen
- checks out the proper decay weights for WW and ZZ decays, so that they are available in the gridpack (for recent gridpacks users had to unpack, add the weights and repack)
- also fixes the way the JHU input file is handles. Previous version of the code was deleting JHUGen.input before doing anything with it.

Fixes for HJ + NNLOPS:
- fixes the input file name for nnlops reweighter 
- fixes a problem running HJ + CT14 (code was trying to access member 1 for alphas variation, when there is no member 1 for those variations, only member 0 in LHAPDF 6.1.6)

perrozzi commented on bin/Powheg/run_pwg.py:
```diff
@@ -329,6 +329,10 @@ def runGetSource(parstage, xgrid, folderName, powInputName, process, tagName) :
 if [[ -s ./JHUGen.input ]]; then
   jhugen=$(expr $jhugen + 1)
   echo "JHUGen activated!"
+  #for decay weights in H->WW and H->ZZ
```
naive question: so far we produced already HZZ and HWW gridpacks, why is this addition needed?

perrozzi commented on bin/Powheg/run_pwg.py:
```diff
@@ -968,10 +982,13 @@ def runhnnlo(folderName, njobs, QUEUE):
-        os.system('rm -rf JHUGen.input')
-        inputJHUGen = '/'.join(powInputName.split('/')[0:-1])+'/JHUGen.input'
+        #os.system('rm -rf JHUGen.input')
+        #inputJHUGen = '/'.join(powInputName.split('/')[0:-1])+'/JHUGen.input'
```
was this really an empty string before JHUGen.input ?!?

perrozzi replied:
can you please restore this, according to yuan chao's comment?

lenzip replied:
It should be done now.
Let me know if you are able to merge now.
Thanks!
Giulio

lenzip commented:
As far as I know people used to unpack the gridpack, add this stuff and 
repack it.
Now it should end up in the gridpack automatically.
Cheers
Giulio

On 06/10/2016 11:42, perrozzi wrote:
>
> *@perrozzi* commented on this pull request.
>
> ------------------------------------------------------------------------
>
> In bin/Powheg/run_pwg.py 
> <https://github.com/cms-sw/genproductions/pull/966#pullrequestreview-3077790>:
>
> > @@ -329,6 +329,10 @@ def runGetSource(parstage, xgrid, folderName, powInputName, process, tagName) :
>   if [[ -s ./JHUGen.input ]]; then
>     jhugen=$(expr $jhugen + 1)
>     echo "JHUGen activated!"
> +  #for decay weights in H->WW and H->ZZ
>
> naive question: so far we produced already HZZ and HWW gridpacks, why 
> is this addition needed?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/cms-sw/genproductions/pull/966#pullrequestreview-3077790>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AErgVoxCJv98qpBBJuJGvkkEB5bma01Tks5qxMKdgaJpZM4KOtET>.
>

lenzip commented:
I did not quite understand this.
first thing it did was to remove the user provided  JHUGen.input and 
then try to download a generic one.
If others understand how this was supposed to work I'm happy to change 
it as needed.
Cheers
Giulio

On 06/10/2016 11:44, perrozzi wrote:
>
> *@perrozzi* commented on this pull request.
>
> ------------------------------------------------------------------------
>
> In bin/Powheg/run_pwg.py 
> <https://github.com/cms-sw/genproductions/pull/966#pullrequestreview-3078098>:
>
> > @@ -968,10 +982,13 @@ def runhnnlo(folderName, njobs, QUEUE):
>           os.system('mkdir -p '+rootfolder+'/'+args.folderName)
>           os.system('cp -p '+args.inputTemplate.split('/')[-1]+' '+args.folderName+'/powheg.input')
>   
> -        os.system('rm -rf JHUGen.input')
> -        inputJHUGen = '/'.join(powInputName.split('/')[0:-1])+'/JHUGen.input'
> +        #os.system('rm -rf JHUGen.input')
> +        #inputJHUGen = '/'.join(powInputName.split('/')[0:-1])+'/JHUGen.input'
>
> was this really an empty string before JHUGen.input ?!?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/cms-sw/genproductions/pull/966#pullrequestreview-3078098>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AErgVvYguRUIFncsk4XbU2z_2F-7rCmgks5qxMMCgaJpZM4KOtET>.
>

perrozzi commented:
@yuanchao  could you please check these modifications wrt yours?
